crates/subspace-networking/src/behavior.rs

```diff
@@ -1,18 +1,55 @@
-use crate::kad::custom_record_store::CustomRecordStore;
-use libp2p::identify::{Identify, IdentifyEvent};
-use libp2p::kad::{Kademlia, KademliaEvent};
+pub(crate) mod custom_record_store;
+
+use crate::create::ValueGetter;
+use custom_record_store::CustomRecordStore;
+use libp2p::identify::{Identify, IdentifyConfig, IdentifyEvent};
+use libp2p::kad::{Kademlia, KademliaConfig, KademliaEvent};
 use libp2p::ping::{Ping, PingEvent};
-use libp2p::NetworkBehaviour;
+use libp2p::{Multiaddr, NetworkBehaviour, PeerId};
+
+pub(crate) struct BehaviorConfig {
+    /// Identity keypair of a node used for authenticated connections.
+    pub(crate) peer_id: PeerId,
+    /// Nodes to connect to on creation.
+    pub(crate) bootstrap_nodes: Vec<(PeerId, Multiaddr)>,
+    /// The configuration for the [`Identify`] behaviour.
+    pub(crate) identify: IdentifyConfig,
+    /// The configuration for the [`Kademlia`] behaviour.
+    pub(crate) kademlia: KademliaConfig,
+    /// Externally provided implementation of value getter for Kademlia DHT,
+    pub(crate) value_getter: ValueGetter,
+}
 
 #[derive(NetworkBehaviour)]
 #[behaviour(out_event = "Event")]
 #[behaviour(event_process = false)]
-pub(crate) struct Behaviour {
+pub(crate) struct Behavior {
```

Comment:
I'm curious why choose `Behavior` over `Behaviour`.

Reply:
I'm more used to en-us rather than en-gb, this is why the file didn't match the struct name, I decided in order to avoid doing typos I can as well rename the struct going forward.